### PR TITLE
Bnb/mirror_padding

### PR DESCRIPTION
**Pull request by bnb32**
added ReflectionPadding class to custom_layers.py. issue #27

### CODE REVIEW

**Comment by grantbuster** [comment]
Looks good! Can you just add a test then we'll merge?

**Comment by grantbuster** [comment]
Not sure whats up with the tf1.15 test. I can take care of that later.

**Comment by bnb32** [comment]
yeah it looks like a file for that test was removed during the refactoring

**Review by grantbuster — CHANGES_REQUESTED** [review]
A few changes, mostly on the test methods. also, in `.github/workflows/pull_request_tests_tf_1_15.yml` can you try changing `python -m pytest ./tests/test_tf_layers.py -v` to ->> `python -m pytest ./tests/test_layers.py -v` ? I think this is the only reason that the pytests are failing.

File: tests/test_layers.py [hunk 1]
```diff
@@ -6,10 +6,53 @@
 import pytest
 import tensorflow as tf
 
+from phygnn.layers.custom_layers import FlexiblePadding
 from phygnn.layers.custom_layers import SkipConnection, SpatioTemporalExpansion
 from phygnn.layers.handlers import Layers, HiddenLayers
 
 
+def test_padding_layers():
```
[the remark this answers is not in the document]
Comment (bnb32) [reply tests/test_layers.py:14]: Yeah I was unsure how exactly to "integrate with hiddenlayers". This makes sense.

File: tests/test_layers.py [hunk 2]
```diff
@@ -147,3 +190,31 @@ def test_st_expansion_bad():
...
+@pytest.mark.parametrize(
+    ('paddings', 'mode'),
+    (([[1, 1], [2, 2]], 'REFLECT'),
+     ([[1, 1], [2, 2]], 'CONSTANT'),
+     ([[1, 1], [2, 2]], 'SYMMETRIC')))
+def test_flexible_padding(paddings, mode):
+    """Test flexible padding routine"""
+    layer = FlexiblePadding(paddings, mode)
+    t = tf.constant([[1, 2, 3],
+                     [4, 5, 6]])
```
[the remark this answers is not in the document]
Comment (bnb32) [reply tests/test_layers.py:204]: Sounds good!

File: tests/test_layers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (grantbuster) [reply tests/test_layers.py:14]: I should have explained this better, basically the `HiddenLayers` class is a handler that takes a config-style set of arguments to make an ordered list of network layers. You can retrieve any of the custom layer classes by setting `'class': CustomClass`. There's more detail in the `HiddenLayers` docstrings too.

File: phygnn/layers/custom_layers.py [hunk 3]
```diff
@@ -3,10 +3,54 @@
...
+
+    def call(self, x):
+        """calls the padding routine
+
+        Parameters
+        ----------
+        x : tf.Tensor
+            tensor on which to perform padding
+        """
+        return tf.pad(x, tf.constant(self.paddings),
```
[the remark this answers is not in the document]
Comment (bnb32) [reply phygnn/layers/custom_layers.py:50]: good point

File: tests/test_layers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bnb32) [reply tests/test_layers.py:14]: Yeah the example helped. Thanks!

**Review by grantbuster — APPROVED** [review]

**Pull request merged**